<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>860</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>                                                                        Nonce Calculation</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>blockchain.png</normaloff>blockchain.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QFrame" name="frame_block">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>901</width>
      <height>221</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>1</number>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>861</width>
       <height>181</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout__block">
      <item row="2" column="6">
       <widget class="QLabel" name="label_merkleRootHint">
        <property name="text">
         <string>64 hex numbers</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_timestamp">
        <property name="text">
         <string>Timestamp</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="5">
       <widget class="QLineEdit" name="lineEdit_timestamp">
        <property name="maxLength">
         <number>19</number>
        </property>
       </widget>
      </item>
      <item row="3" column="6">
       <widget class="QLabel" name="label_timestampHint">
        <property name="text">
         <string>yyyy-MM-dd HH:mm:ss</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_bits">
        <property name="text">
         <string>Bits</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QRadioButton" name="radioButton_version2">
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_version">
        <property name="text">
         <string>Version</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="radioButton_version1">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1" colspan="5">
       <widget class="QLineEdit" name="lineEdit_difficulty">
        <property name="maxLength">
         <number>12</number>
        </property>
       </widget>
      </item>
      <item row="5" column="6">
       <widget class="QLabel" name="label_difficultyHint">
        <property name="text">
         <string>usually big number</string>
        </property>
       </widget>
      </item>
      <item row="4" column="6">
       <widget class="QLabel" name="label_bitsHint">
        <property name="text">
         <string>decimal with 10 max digits</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="5">
       <widget class="QLineEdit" name="lineEdit_bits">
        <property name="maxLength">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_difficulty">
        <property name="text">
         <string>Difficulty</string>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QLabel" name="label_versionHint">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_previousHash">
        <property name="text">
         <string>Previous Hash</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_merkleRoot">
        <property name="text">
         <string>Merkle Root</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="5">
       <widget class="QLineEdit" name="lineEdit_previousHash">
        <property name="maxLength">
         <number>64</number>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <widget class="QLabel" name="label_previousHashHint">
        <property name="minimumSize">
         <size>
          <width>163</width>
          <height>22</height>
         </size>
        </property>
        <property name="text">
         <string>64 hex numbers</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="5">
       <widget class="QLineEdit" name="lineEdit_merkleRoot">
        <property name="maxLength">
         <number>64</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_confirm">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Confirm</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_zeroes">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>420</y>
      <width>321</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>     Select number of leading zeroes in test target    </string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_zeroes">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>420</y>
      <width>37</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>0</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>6</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_target">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>460</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Target</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_findRealNonce">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>500</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Find Real Nonce</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_nonce">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>540</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Nonce</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_nonce">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>540</y>
      <width>125</width>
      <height>22</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_elapsedTime">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>540</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Elapsed Time</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_elapsedTime">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>540</y>
      <width>221</width>
      <height>22</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_realCalc">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>400</y>
      <width>141</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Real Calculation</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_testCalc">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Test with less target</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_output">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>580</y>
      <width>901</width>
      <height>261</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clearOutput">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>540</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear output</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_reset">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>360</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_findTestNonce">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>500</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Find Test Nonce</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_target">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>460</y>
      <width>561</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="maxLength">
     <number>64</number>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>621</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="2" column="3">
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="3">
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="lineEdit_hashrate">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="pushButton_enterManually">
       <property name="text">
        <string>Enter manually</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_block">
       <property name="maxLength">
        <number>7</number>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_blockNumber">
       <property name="text">
        <string>Block Number</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_hashrate">
       <property name="text">
        <string>SHA256 Hashrate</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1" alignment="Qt::AlignHCenter">
      <widget class="QLabel" name="label_blockchainInfo">
       <property name="text">
        <string>From https://blockchain.info</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="2" alignment="Qt::AlignLeft">
      <widget class="QPushButton" name="pushButton_getBlockInformation">
       <property name="text">
        <string>Get Block Information</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_hashrateText">
       <property name="maximumSize">
        <size>
         <width>280</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Average hashing speed with software </string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
